mono/mini/aot-runtime.c

```diff
@@ -2079,6 +2079,25 @@ init_amodule_got (MonoAotModule *amodule, gboolean preinit)
 	mono_loader_unlock ();
 }
 
+#ifdef MONOTOUCH
```

Comment:
should this be ` && TARGET_ARM` so we only use this for ARM and not ARM64 (which is the majority of devices)?

Reply:
Something similar is done here: https://github.com/imhameed/mono/blob/e1d3b7ba6440922cb0addbc26dc72f4788455758/mono/mini/aot-runtime.c#L5241

ld64 has support for emitting branch islands on ARM64. The reason the ARM64 "dont link" test doesn't cause the linker to generate branch islands is because ARM64 unconditional branches support larger displacements (documented as +-128MB, encoded as a 26-bit signed immediate), and the maximum distance between any two executable symbols in "dont link" is about 45.5MB.

One thing I don't like about this patch currently is that `mono_arch_get_call_target` will follow both `b` and `bl`, but when following a chain of branch island jumps, we really should only consider `b`. I could add a new function that disassembles only `b` instructions, and update both branch island 'resolvers'. But maybe this doesn't matter in practice.